### DISCUSSION TITLE
Flutter and Dart sdks should support ranges

### DISCUSSION THREAD

**Discussion by KoheiKanagu** [renovatebot/renovate#26346] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

I think there is a bug in the implementation that extracts the Flutter and Dart versions from pubspec.lock.

The Flutter and Dart versions are listed as follows.
```lock
sdks:
  dart: ">=3.3.0-174.3.beta <4.0.0"
  flutter: ">=3.18.0-0"
````

but renovate is trying to resolve the version with semver and failing.
So, it fails to update the lock file.

I think the correct versioning is npm.

Repositories that can reproduce the problem: https://github.com/KoheiKanagu/renovate_pub_issue

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: Using containerbase dynamic installs (branch="renovate/dart-3.x")
WARN: Invalid tool constraint (branch="renovate/dart-3.x")
{
  "toolName": "flutter"
  "constraint": ">=3.3.0"
  "versioning": "semver"
}

DEBUG: Resolved stable matching version (branch="renovate/dart-3.x")
{
  "toolName": "flutter"
  "resolvedVersion": "3.16.4"
}

DEBUG: Executing command (branch="renovate/dart-3.x")
{
  "command": "install-tool flutter 3.16.4"
}

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
Converted to bug issue: #26349
